### DISCUSSION TITLE
Adding field to feature layer?

### DISCUSSION THREAD

**Discussion by sarahhurley** [R-ArcGIS/arcgislayers#177] [General]
Hi there - I am wondering if there is the capability of adding a field to a feature layer through the arcgislayers package? I'm working on setting up a workflow that may require us to add a field to a dataset over time (trying to avoid needing to republish it every time). I've tried to look through the documentation, but I've only found in the "[Overwrite Hosted Feature Layer](https://r.esri.com/r-bridge-site/location-services/workflows/overwrite-feature-service.html)" section of the R-ArcGIS Bridge documentation a line that says "Truncating a service does not change the field definitions or permit us to add or remove fields to it. If you wish to do so, publish a new layer instead."

Thanks for your help!

**Comment by JosiahParry** [comment]
This is not possible from arcgislayers today. It can be done manually if you are crafty enough with {httr2} ! 
REST documentation: https://developers.arcgis.com/rest/services-reference/online/add-to-definition-feature-layer-.htm

I'm going to convert this to an issue as a feature request. 

A quick note on this capability, though, is that only the **owner** or **admin** can perform this operation

**Reply by sarahhurley** [reply]
Okay - that is good to know, thanks so much!
